core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -344,6 +344,9 @@ public int from() {
      * The number of search hits to return. Defaults to <tt>10</tt>.
      */
     public SearchSourceBuilder size(int size) {
+        if (size < 0) {
+            throw new IllegalArgumentException("[size] parameter cannot be negative");
+        }
```

Comment:
I think it's best to keep the default as-is here and let it be handled when creating the context as today so that we can distinguish between set and set from default.